whisper/whisperv6/whisper.go

```diff
@@ -850,9 +981,16 @@ func validatePrivateKey(k *ecdsa.PrivateKey) bool {
 	return ValidatePublicKey(&k.PublicKey)
 }
 
-// validateSymmetricKey returns false if the key contains all zeros
-func validateSymmetricKey(k []byte) bool {
-	return len(k) > 0 && !containsOnlyZeros(k)
+// validateSymmetricKey returns false if the key contains all zeros,
+// which is a simplest and the most common bug.
+func validateRandomData(k []byte, expectedSize int) bool {
```

Comment:
name is not up to date in the comment, and that function name is really bad imho. It sounds like it validate data that is generated at random, which is presumably not what it is. This is a function that checks that the length of the key is compatible with some scheme I'm still trying to understand at this point. The name should reflect that.

Reply:
The purpose of this function is to check, if generated random data (used for key, salt or padding) is indeed filled with something, and not with zeros. There is no way to really check the true randomness, but the simplest and the most common bug is just using zeros without any checks. Sounds ridiculous, should never happen, but this bug has already sent thousands of Ether to address zero. 

However, zero is quite legitimate value of any randomly generated byte. Recently I have noticed that a test failed after trying to generate a random padding of size 1, because it happend to be zero. So, I increased the threshold to 4 bytes. We need 12 bytes for Salt and 32 bytes for Key anyway, so it will only affect the small-sized padding, which is not critical anyways.

Concerning the name -- it is no good indeed. Please suggest a better one. If you don't like the comment, please suggest a better one. I would really appreciate your help here.